src/endpoint.js

```diff
@@ -1,7 +1,8 @@
 export default class Endpoint {
-    constructor(httpClient, basePath) {
+    constructor(httpClient, basePath, requestOptions) {
```

Comment:
Hmm, thinking about it. The class should not be `Endpoint` but `EndpointGroup` or `Resource` because endpoints are the individual functions.

Then the options would be called `endpointGroupOptions` or `resourceOptions`.

I think I like `Resource` the best.